test/integration/tests/button/message.js

```diff
@@ -0,0 +1,408 @@
+/* @flow */
+/* eslint max-lines: 0 */
+
+import { wrapPromise, getElement } from "@krakenjs/belter/src";
+import { FUNDING } from "@paypal/sdk-constants/src";
+
+import { CLASS } from "../../../../src/constants";
+import {
+  assert,
+  getElementRecursive,
+  createTestContainer,
+  destroyTestContainer,
+} from "../common";
+
+describe(`paypal button message`, () => {
+  beforeEach(() => {
+    createTestContainer();
+  });
+
+  afterEach(() => {
+    destroyTestContainer();
+  });
+
+  describe("reserves space for message", () => {
+    it("should reserve space for a message when messageMarkup is undefined", (done) => {
+      window.paypal
+        .Buttons({
+          message: {},
+          test: {
+            onRender() {
+              const frame = document.querySelector("#testContainer iframe");
+              if (!frame) {
+                throw new Error(`Cannot find frame`);
+              }
+              // $FlowFixMe
+              const win = frame.contentWindow;
+              const message = win.document.querySelector(
+                ".paypal-button-message"
+              );
+              if (!message) {
+                done(new Error("No message generated"));
+              }
+              const reservationDiv = win.document.querySelector(
+                ".paypal-button-message-reserved"
+              );
+              if (!reservationDiv) {
+                done(new Error("message space was not reserved"));
+              }
+              done();
+            },
+          },
+        })
+        .render("#testContainer");
+    });
+
+    it("should not reserve space for a message when messageMarkup is an empty string", (done) => {
+      window.paypal
+        .Buttons({
+          message: {},
+          messageMarkup: "",
+          test: {
+            onRender() {
+              const frame = document.querySelector("#testContainer iframe");
+              if (!frame) {
+                throw new Error(`Cannot find frame`);
+              }
+
+              // $FlowFixMe
+              const win = frame.contentWindow;
+              const message = win.document.querySelector(
+                ".paypal-button-message"
+              );
+              if (!message) {
+                done(new Error("No message generated"));
+              }
+              const reservationDiv = win.document.querySelector(
+                ".paypal-button-message-reserved"
+              );
+              if (reservationDiv) {
+                done(new Error("message space was reserved incorrectly"));
+              }
+              done();
+            },
+          },
+        })
+        .render("#testContainer");
+    });
+
+    it("should not reserve space for a message when messageMarkup is truthy", (done) => {
```

Comment:
Should this be `...when messageMarkup is string` since we have a `typeof` string check as the decision?

Reply:
I wanted to have tests that differentiate these cases:

1. messageMarkup is `""`
2. messageMarkup is a string with length > 0.

Would those labels be better?